include/orng.hpp

```diff
@@ -79,10 +80,22 @@ CONTRACT orng: public eosio::contract {
      * 
      * @param exponent The public key exponent
      * @param modulus The public key modulus
+     * @note it uses the integer of hash modulus as a table scope
      */
-    ACTION setsigpubkey(const std::string& exponent, const std::string& modulus);
+    ACTION setsigpubkey(uint64_t id, const std::string& exponent, const std::string& modulus);
     using setsigpubkey_action = eosio::action_wrapper<"setsigpubkey"_n, &orng::setsigpubkey>;
 
+    /**
+    * @dev clean the signing values from dapp which has been signed with no longer used public-key.
+    * @param pubkey_id the id of public-key.
+    * @param rows_num The number of rows that be expected to be removed
+    * @note it does not allow to removing the signing values which have scope is the id of active public-key
+    */
+    ACTION cleansigvals(uint64_t pubkey_id, uint64_t rows_num);
+    using cleansigvals_action = eosio::action_wrapper<"cleansigvals"_n, &orng::cleansigvals>;
+
+    ACTION setduration(uint64_t duration);
```

Comment:
I think it will be less gameable and more trustless if instead we did something like this [here](https://github.com/worldwide-asset-exchange/wax-orng/blob/develop/src/orng.cpp#L117)
```
if (rv_hash % 1000000 == 0)  {  // 1000000 could be a config value for the chance of switching out the signing key - see `chance_to_switch` in my comment about this config table
  switch_to_next_signing_key()
}
```